### PR TITLE
Making comments after connexion

### PR DESCRIPTION
**Pull request by tanguyguillo**
users until there are connected have to have the possibility to let their comments for an article ... it's not the case now .... have to do

### CODE REVIEW

File: app/controllers/CommentController.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            $this->twig->display('info/info.html.twig', compact('message'));
+            exit;
+        }
+
+        //$arrayComment = $arrayComment;
+        $connection = new DatabaseConnection();
+        $Comment = new Comment();
+        $Comment->connection = $connection;
+
+        if ($Comment->SetComments($arrayComment)) {
```
Comment (castroCrea) [inline app/controllers/CommentController.php:41]:
```suggestion
        if ($Comment->setComments($arrayComment)) {
```

File: app/controllers/CommentController.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        // // if your not connected you can't send a comment
+        if (($_SESSION['LOGGED_USER_NAME'] == "")) {
+            $message = "Vous devez être connecté pour pouvoir rédiger un commentaire";
+            $this->twig->display('info/info.html.twig', compact('message'));
+            exit;
+        }
+
+        //$arrayComment = $arrayComment;
+        $connection = new DatabaseConnection();
+        $Comment = new Comment();
```
Comment (castroCrea) [inline app/controllers/CommentController.php:38]:
```suggestion
        $comment = new Comment();
```

File: app/controllers/DetailController.php [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    //     $this->test = 22;
+    // }
+
+    /**
+     * Function to show one blog post
+     *
+     * @param string $identifier
+     * @return void
+     */
+    public function Detail($identifier, $message = '')
```
Comment (castroCrea) [inline app/controllers/DetailController.php:30]:
```suggestion
    public function detail($identifier, $message = '')
```

File: app/controllers/DetailController.php [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+     *  function to get all data which is useful - EXAMPLE  string(10) "tsd@fqd.fr" string(21) "sffqddfdqfqdfdqsffdqs"
+     * htmlspecialchars($_COOKIE["name"])
+     * time() + 365 * 24 * 3600 : one year
+     * time() + 3600 : one hour ?
+     *
+     * @param [array] $postData
+     * @param [string] $messsage
+     * @return void
+     */
+    public function DetailConnexion(array $postData, string $messsage = '')
```
Comment (castroCrea) [inline app/controllers/DetailController.php:89]:
```suggestion
    public function detailConnexion(array $postData, string $messsage = '')
```

File: app/controllers/DetailController.php [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+    {
+        // if exist
+        if (isset($postData['user_login']) &&  isset($postData['user_pass'])) {
+            $messsageReadle = ""; // for no data message
+
+            $connection = new DatabaseConnection();
+            $UsersRepository = new UsersRepository();
```
Comment (castroCrea) [inline app/controllers/DetailController.php:96]:
```suggestion
            $usersRepository = new UsersRepository();
```

Comment (castroCrea) [reply app/controllers/DetailController.php:96]: constant -> camelCase

File: app/controllers/InscriptionController.php [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  {
+    $this->twig->display('inscription/inscription.html.twig');
+  }
+
+  /**
+   * function to create a user account (with crypt password)
+   *
+   * @return void
+   */
+  public function CreateAccount(array $postData)
```
Comment (castroCrea) [inline app/controllers/InscriptionController.php:32]:
```suggestion
  public function createAccount(array $postData)
```

function -> camelCase

File: app/model/CommentModel.php [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        return $postComments;
+    }
+
+    /**
+     * function to write a comment
+     *
+     * @param array
+     * @return void
+     */
+    public function SetComments(array $Array)
```
Comment (castroCrea) [inline app/model/CommentModel.php:58]:
```suggestion
    public function setComments(array $Array)
```

File: app/model/CommentModel.php [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+            // sometimes it's happens ... isuue with data refreshed page
+            if ($user_id == 0) {
+                $user_id = $_SESSION['LOGGED_USER_ID'];
+                if ($user_id == 0) {
+                    return false;
+                }
+            }
+
+            $CommentCreated = date('Y-m-d h:i:s');
```
Comment (castroCrea) [inline app/model/CommentModel.php:71]:
```suggestion
            $commentCreated = date('Y-m-d h:i:s');
```

File: app/model/CommentModel.php (hunk 7 above)
Comment (castroCrea) [inline app/model/CommentModel.php:58]:
```suggestion
    public function SetComments(array $array)
```

File: app/Core/Database/database.php [hunk 9]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Application\Core\Database\Database;
+
+// Loading the config
+require_once(ROOT . '/app/config/config.php');
+
+class DatabaseConnection
```
Comment (castroCrea) [inline app/Core/Database/database.php:8]: nom du fichier DatabaseConnection

**Review by castroCrea — COMMENTED** [review]
juste des typos

File: app/Core/Database/database.php (hunk 9 above)
Comment (tanguyguillo) [reply app/Core/Database/database.php:8]: to change the file's name to DatabaseConnection making the application no more working... I keep it on mind... i'll return on it when i'll have a clearest brain. worked on my actuel branch Making-adminAdmin and i do a commit for all the others camelCase issues...
thanks a lot for all this work of re reading's code

Comment (tanguyguillo) [reply app/Core/Database/database.php:8]: Ok it's done : 

Capture d’écran 2022-08-01 à 11 57 53

I named the file DatabaseConnexion to stay in the Englishspoken logic...

**Comment by tanguyguillo** [comment]
For me this part is finished @castroCrea

**Review by castroCrea — APPROVED** [review]

**Pull request closed without merging**